clients/privacy-center/cypress/e2e/banner-overlay-dismissal.cy.ts

```diff
@@ -1,309 +1,173 @@
-import { ConsentMethod } from "fides-js";
+import { ConsentMethod, CONSENT_COOKIE_NAME, FidesCookie } from "fides-js";
 import { stubConfig } from "../support/stubs";
 
-function assertDismissCalled() {
-  cy.get("@FidesUpdated")
-    .should("have.been.calledOnce")
-    .its("lastCall.args.0.detail.extraDetails.consentMethod")
-    .then((consentMethod) => {
-      expect(consentMethod).to.eql(ConsentMethod.dismiss);
-    });
-
-  cy.wait("@patchPrivacyPreference");
-}
-
-describe("Banner and overlay dismissal", () => {
-  describe("Consent with dismissal allowed", () => {
-    beforeEach(() => {
-      stubConfig({
-        options: {
-          tcfEnabled: false,
-          preventDismissal: false,
-        },
+describe("Banner and modal dismissal", () => {
+  // Helper function for some test case assertions
+  function assertDismissCalled() {
+    cy.get("@FidesUpdated")
+      .should("have.been.calledOnce")
+      .its("lastCall.args.0.detail.extraDetails.consentMethod")
+      .then((consentMethod) => {
+        expect(consentMethod).to.eql(ConsentMethod.DISMISS);
+      });
+
+    cy.waitUntilCookieExists(CONSENT_COOKIE_NAME).then(() => {
+      cy.getCookie(CONSENT_COOKIE_NAME).then((cookie) => {
+        const fidesCookie: FidesCookie = JSON.parse(
+          decodeURIComponent(cookie!.value)
+        );
+        expect(fidesCookie.fides_meta)
+          .property("consentMethod")
+          .is.eql(ConsentMethod.DISMISS);
       });
     });
 
-    describe("Banner dismissal", () => {
-      it("Should dismiss the banner by clicking the x", () => {
-        cy.get("#fides-banner").should("be.visible");
-        cy.get("#fides-banner").within(() => {
-          cy.get(".fides-close-button").click();
+    cy.wait("@patchPrivacyPreference");
+  }
+
+  // Test all combinations of TCF enabled/disabled and prevent dismissal enabled/disabled
+  interface TestCaseOptions {
+    tcfEnabled: boolean;
+    preventDismissal: boolean;
+  }
+
+  const testCases: TestCaseOptions[] = [
+    { tcfEnabled: false, preventDismissal: false },
+    { tcfEnabled: false, preventDismissal: true },
+    { tcfEnabled: true, preventDismissal: false },
+    { tcfEnabled: true, preventDismissal: true },
+  ];
+
+  testCases.forEach(({ tcfEnabled, preventDismissal }) => {
```

Comment:
Mainly for @galvana: for myself, I refactored this to be a parameterized test because I know I want to add a few more assertions to this soon!

Contents of this are _mostly_ unchanged though.

Reply:
Nice! Thanks for the cleanup here